platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/cache/CookieCache.java

```diff
@@ -58,9 +58,7 @@ public void addActiveSp(String key, String activeSp) {
             LOGGER.error("Cannot add sp [{}] for [{}]: does not exist", activeSp, key);
             return;
         }
-        synchronized (dataWrapper) {
```

Comment:
I concur, you should add it to the get and remove as well. If we were only synchronizing on the SP Set itself, then I would say no.